### DISCUSSION TITLE
Direction labels and selected directions

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#5809] [Ideas]
The direction dialer, that shows up when clicking the direction box in the information area, is cool but it is little informative.

My proposal is:

- adding to the dialer direction labels for _common directions_ (e.g. 0, 45, 90, 135, 180, 225, 270, 315 or 0, 45, 90, 135, 180, -135, -90, -45)
- allowing the sprite to point only to the common directions when holding the shift key while rotating the direction arrow

**Comment by Samq64** [comment]
> - allowing the sprite to point only to the common directions when holding the shift key while rotating the direction arrow

See #3699 and #3726.

**Comment by DNin01** [comment]
It doesn’t look like we’re going to be doing the second one - #3726 was closed because it wasn’t useful enough to become an addon.

**Reply by Samq64** [reply]
I think it could be reconsidered.

**Reply by DNin01** [reply]
I think we could repurpose that idea too. Maybe snap to 15° increments without needing to hold Shift, like the point-in-direction block's rotation thingy.

**Reply by sfederici** [reply]
As suggested in today's updates to https://github.com/ScratchAddons/ScratchAddons/pull/3726, the free rotation in the right dial could be useful.

**Reply by DNin01** [reply]
> As suggested in today's updates to https://github.com/ScratchAddons/ScratchAddons/pull/3726, the free rotation in the right dial could be useful.

What if we inverted the controls: holding Shift would allow free rotation but otherwise it would snap?

**Comment by sfederici** [comment]
What I suggest is snapping (in the right dial) when holding shift. Indeed, for the Stage, the naive user would expect to be able to freely rotate the sprite. So, this should be the default behaviour.
